src/ModelContextProtocol.Core/Protocol/JsonRpcMessage.cs

```diff
@@ -38,6 +39,19 @@ private protected JsonRpcMessage()
     [JsonIgnore]
     public ITransport? RelatedTransport { get; set; }
 
+    /// <summary>
```

Comment:
Not related to this immediate change, but it feels kind of strange that we're attaching a whole bunch of state to what is otherwise just a model for a protocol message. It does suggest that in the future we might want to refactor things somewhat, either by forcing separation between "domain models" and protocol version specific DTOs or perhaps by passing a context object alongside messages.